rpc/src/main/scala/WorkerServer.scala

```diff
@@ -74,30 +90,76 @@ class WorkerServer(executionContext: ExecutionContext) { self =>
       val promise = Promise[PartitionReply]
 
       Future {
-        // TODO: Logic
-        promise.success(new PartitionReply())
+        try {
+          val block = Block.fromPath(Path("data/block"), 10, 90)
+          request.workers
+            .map(workerMetadata => {
+              val keyRange = KeyRange(
+                Key.fromString(workerMetadata.keyRange.get.from.toStringUtf8),
+                Key.fromString(workerMetadata.keyRange.get.to.toStringUtf8)
+              )
+              val partition = block.partition(keyRange)
```

Comment:
Before making Partition, we have to sort the 'block' so that each partition has sorted Records

Reply:
I think so too, I think it need to be added at the first time worker deals with the `block`

```scala
val sortedBlock = Block.fromPath(Path("data/block"), 10, 90).sort()
sortedBlock.writeTo(Path("data/block"))
```